File: server/lib/script/import.js
import axios from 'axios';
import { JSDOM } from 'jsdom';
import { PrismaClient } from "@prisma/client";
const prisma = new PrismaClient();

const url = 'https://github.com/blakeblackshear/frigate/issues?page=4&q=is%3Aissue+is%3Aopen';

async function insertIssues(url) {
    try {
        const response = await axios.get(url);
        const dom = new JSDOM(response.data);
        const issueElements = dom.window.document.querySelectorAll('.js-issue-row');

        for (const issueElement of issueElements) {
            // Your existing logic here
            const issueNameElement = issueElement.querySelector('.Link--primary.v-align-middle');
            const issueName = issueNameElement ? issueNameElement.textContent.trim() : 'No title found';
            const issueLinkElement = issueElement.querySelector('.Link--primary.v-align-middle');
            const issueLink = issueLinkElement ? 'https://github.com' + issueLinkElement.getAttribute('href') : 'No link found';
            const issueTagsElements = issueElement.querySelectorAll('.IssueLabel');
            let issueTags = Array.from(issueTagsElements).map(tagElement => tagElement.textContent.trim()).join(', ').toLowerCase();
            
            // Assuming you have a logic to create an issue in your database
            if (issueTags){
            await prisma.Issue.create({
                data:{
                    name: issueName,
                    link: issueLink,
                    language: 'python',
                    tag: issueTags
                    
                }
            });
        }

            console.log('Issue Name:', issueName);
            console.log('Issue Link:', issueLink);
            console.log('Issue Tags:', issueTags);
            console.log();
        }
    } catch (error) {
        console.error('Error:', error);
    }
}

insertIssues(url).catch(console.error);

File: server/lib/script/import.py
import requests
from bs4 import BeautifulSoup

# URL of the GitHub repository's issues section
url = "https://github.com/Significant-Gravitas/AutoGPT/issues"

# Send a GET request to the URL
response = requests.get(url)

# Create a BeautifulSoup object with the response content
soup = BeautifulSoup(response.content, "html.parser")

# Find all the issue elements on the page
issue_elements = soup.find_all("div", class_="js-navigation-container js-active-navigation-container")

# Iterate over each issue element and extract the required information
for issue_element in issue_elements:
    # Extract the issue name
    issue_name = issue_element.find("a", class_="Link--primary").text.strip()

    # Extract the issue link
    issue_link = "https://github.com" + issue_element.find("a", class_="Link--primary")["href"]

    # Extract the issue tag
    issue_tag = issue_element.find("span", class_="Label").text.strip()

    # Print the extracted information
    print("Issue Name:", issue_name)
    print("Issue Link:", issue_link)
    print("Issue Tag:", issue_tag)
    print()